pkg/reconciler/finalizer/base_finalizer.go

```diff
@@ -56,3 +67,48 @@ func (bf *baseFinalizer) destroy(ctx context.Context) status.MultiError {
 	klog.Info("Destroyer succeeded")
 	return nil
 }
+
+func (bf *baseFinalizer) unmanageObjects(ctx context.Context, rsyncRef client.ObjectKey) status.MultiError {
+	var errs status.MultiError
+	rg := &v1alpha1.ResourceGroup{}
+	if err := bf.Client.Get(ctx, rsyncRef, rg); err != nil {
+		if apierrors.IsNotFound(err) {
+			// ResourceGroup doesn't exist. This could be for a variety of reasons, but
+			// there's no work for the finalizer to do.
+			return nil
+		}
+		return status.APIServerError(err, "failed to get ResourceGroup", rg)
+	}
+	removeAnnotations := []string{
+		metadata.ResourceManagerKey,
+		metadata.OwningInventoryKey,
+		metadata.ResourceManagementKey,
+	}
+	for _, objRef := range rg.Spec.Resources {
+		// Remove metadata for each object
+		obj := &metav1.PartialObjectMetadata{}
+		obj.SetGroupVersionKind(schema.GroupVersionKind{
+			Group: objRef.Group,
+			Kind:  objRef.Kind,
+		})
+		key := client.ObjectKey{
+			Name:      objRef.Name,
+			Namespace: objRef.Namespace,
+		}
+		if err := bf.Client.Get(ctx, key, obj); err != nil {
+			if apierrors.IsNotFound(err) { // skip orphaning objects that don't exist
+				continue
+			}
+			errs = status.Append(errs, status.APIServerError(err, "failed to get object", obj))
+			continue
+		}
+		existing := obj.DeepCopy()
+		if core.RemoveAnnotations(obj, removeAnnotations...) {
```

Comment:
What about labels?

We have something like this in the Supervisor code already i think, for abandoning individual objects. Might want to make the code similar.

Reply:
The change only removes the management annotations which block the resources from being adopted by another client/RSync (by either the admission-webhook or in the applier logic).

It was intentional to leave the other Config Sync metadata (annotations/labels), but I'm not opposed to removing all of the metadata if you feel strongly

Reply:
I think removing all the CS metadata would be best, unless there's some use case that needs it.